contracts/jobs/JobsManager.sol

```diff
@@ -70,94 +163,300 @@ contract JobsManager is IJobsManager, Verifiable, Controllable {
+
+        // Segment must be eligible for verification
+        require(JobLib.shouldVerifySegment(_segmentNumber, claim.segmentRange, claim.claimBlock, verificationRate));
+        // Receipt must be valid
+        require(validateReceipt(_jobId, _claimId, _segmentNumber, _dataHash, _transcodedDataHash, _broadcasterSig, _proof));
+
+        // Mark segment as submitted for verification
+        claim.segmentVerifications[_segmentNumber] = true;
+
+        // Invoke transcoding verification. This is async and will result in a callback to receiveVerification() which is implemented by this contract
+        verifier.verify(_jobId, _claimId, _segmentNumber, _dataHash, _transcodedDataHash, this);
+
+        return true;
     }
 
     /*
-     * @dev Return stream id for job
+     * @dev Callback function that receives the results of transcoding verification
      * @param _jobId Job identifier
+     * @param _segmentNumber Segment being verified for job
+     * @param _result Boolean result of whether verification succeeded or not
      */
-    function getJobStreamId(uint256 _jobId) public constant returns (string) {
-        return jobs.jobs[_jobId].streamId;
+    function receiveVerification(uint256 _jobId, uint256 _claimId, uint256 _segmentNumber, bool _result) onlyVerifier external returns (bool) {
+        if (!_result) {
+            refundBroadcaster(_jobId, _claimId);
+            // Protocol slashes transcoder for failing verification (no finder)
+            bondingManager().slashTranscoder(jobs[_jobId].transcoderAddress, address(0), failedVerificationSlashAmount, 0);
+        }
+
+        ReceivedVerification(_jobId, _claimId, _segmentNumber, _result);
+
+        return true;
     }
 
     /*
-     * @dev Return transcoding options for job
+     * @dev Distribute fees for multiple claims
      * @param _jobId Job identifier
+     * @param _claimId Claim identifier
      */
-    function getJobTranscodingOptions(uint256 _jobId) public constant returns (string) {
-        return jobs.jobs[_jobId].transcodingOptions;
+    function batchDistributeFees(uint256 _jobId, uint256[] _claimIds) external returns (bool) {
+        for (uint256 i = 0; i < _claimIds.length; i++) {
+            distributeFees(_jobId, _claimIds[i]);
+        }
+
+        return true;
     }
 
     /*
-     * @dev Return transcode claims details for a job
+     * @dev Slash transcoder for missing verification
      * @param _jobId Job identifier
+     * @param _claimId Claim identifier
+     * @param _segmentNumber Segment that was not verified
      */
-    function getJobTranscodeClaimsDetails(uint256 _jobId) public constant returns (uint256, uint256, uint256, uint256, bytes32) {
-        return jobs.getJobTranscodeClaimsDetails(_jobId);
+    function missedVerificationSlash(uint256 _jobId, uint256 _claimId, uint256 _segmentNumber) external returns (bool) {
+        Job storage job = jobs[_jobId];
+        Claim storage claim = job.claims[_claimId];
```

Comment:
I believe this is an index into an array, and therefore will throw if the claim doesn't exist. Just sanity checking that since we're not explicitly checking this returns a valid claim (if it were a mapping I think we'd have to check.)